src/hooks/core/post/usePostHook.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import {Context} from '../../../context';
+import {setMainFeeds, setTopicFeeds} from '../../../context/actions/feeds';
+import {followUserAnon, setFollow, setUnFollow, unfollowUserAnon} from '../../../service/profile';
+import removePrefixTopic from '../../../utils/topics/removePrefixTopic';
+import TopicPageStorage from '../../../utils/storage/custom/topicPageStorage';
+
+const usePostHook = () => {
+  const [profileContext] = (React.useContext(Context) as any).profile;
+  const [feedsContext, feedsContextDispatch] = (React.useContext(Context) as any).feeds;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The context is being cast to `any` which can lead to potential type errors. It would be better to define a type for the context and use that instead of `any`.
```diff
-  const [profileContext] = (React.useContext(Context) as any).profile;
-  const [feedsContext, feedsContextDispatch] = (React.useContext(Context) as any).feeds;
+  const [profileContext] = (React.useContext(Context) as ProfileContextType).profile;
+  const [feedsContext, feedsContextDispatch] = (React.useContext(Context) as FeedsContextType).feeds;
```
Please replace `ProfileContextType` and `FeedsContextType` with the actual types you have defined for your context.



<!-- This is an auto-generated comment by OSS CodeRabbit -->